encoders/src/main/scala/au/csiro/pathling/encoders/datatypes/CustomCoder.scala

```diff
@@ -12,15 +12,23 @@
 
 package au.csiro.pathling.encoders.datatypes
 
+import au.csiro.pathling.encoders2.ExpressionWithName
 import org.apache.spark.sql.catalyst.expressions.Expression
-import org.apache.spark.sql.types.StructField
+import org.apache.spark.sql.types.{DataType, StructField}
 
 trait CustomCoder {
-
+  @deprecated
   def schema: Seq[StructField]
 
+  @deprecated
   def customDecoderExpression(addToPath: String => Expression): Expression
 
+  @Deprecated
```

Comment:
What's the difference between `@deprecated` and `@Deprecated`?

Reply:
@deprecated is from scala, while @Deprecataed  is from Java.
I made a change to use the scala version consistently.